### PR TITLE
Refactor awsClient

### PR DESCRIPTION
**Pull request by ArielLima**
Refact account controller to use awsClientBuilder when building awsClients.

Ticket: https://issues.redhat.com/browse/OSD-2839
Test:

- Created an account

### CODE REVIEW

**Comment by iamkirkbater** [comment]
/lgtm

Tested with `make test-all` with a local rebase to get the changes from #342.

**Comment by ArielLima** [comment]
rebased

**Comment by rogbas** [comment]
@rogbas please test locally

**Comment by jharrington22** [comment]
Looks good awaiting testing.

File: pkg/controller/account/account_controller.go [hunk 1]
```diff
@@ -106,7 +106,7 @@ type ReconcileAccount struct {
 	// that reads objects from the cache and writes to the apiserver
 	Client           kubeclientpkg.Client
 	scheme           *runtime.Scheme
-	awsClientBuilder func(awsAccessID, awsAccessSecret, token, region string) (awsclient.Client, error)
+	awsClientBuilder func(kubeClient kubeclientpkg.Client, input awsclient.NewAwsClientInput) (awsclient.Client, error)
```
Comment (fahlmant) [inline pkg/controller/account/account_controller.go:109]: Why is the kubeClient being passed here?

**Comment by fahlmant** [comment]
Merge Monday June 1

**Pull request merged**